unittests/eosvmoc_limits_tests.cpp

```diff
@@ -0,0 +1,139 @@
+   chain.set_abi("eosio.token"_n, test_contracts::eosio_token_abi());
+
+   if (chain.control->is_eos_vm_oc_enabled()) {
+      BOOST_CHECK_EXCEPTION(
+         chain.push_action( "eosio.token"_n, "create"_n, "eosio.token"_n, mvo()
+            ( "issuer", "eosio.token" )
+            ( "maximum_supply", "1000000.00 TOK" )),
+         eosio::chain::wasm_execution_error,
+         [](const eosio::chain::wasm_execution_error& e) {
+            return expect_assert_message(e, "failed to compile wasm");
+         }
+      );
+   } else {
+      chain.push_action( "eosio.token"_n, "create"_n, "eosio.token"_n, mvo()
+         ( "issuer", "eosio.token" )
+         ( "maximum_supply", "1000000.00 TOK" )
+      );
+   }
+}
+
+// common routine to verify no wasm_execution_error is raised
+// because limits specified in eosvmoc_config are not reached
+void limit_not_violated_test(const eosvmoc::config eosvmoc_config) {
+   fc::temp_directory tempdir;
+
+   constexpr bool use_genesis = true;
+   validating_tester chain(
+      tempdir,
+      [&](controller::config& cfg) {
+         cfg.eosvmoc_config = eosvmoc_config;
+      },
+      use_genesis
+   );
+
+   chain.create_accounts({"eosio.token"_n});
+   chain.set_code("eosio.token"_n, test_contracts::eosio_token_wasm());
+   chain.set_abi("eosio.token"_n, test_contracts::eosio_token_abi());
+
+   chain.push_action( "eosio.token"_n, "create"_n, "eosio.token"_n, mvo()
+      ( "issuer", "eosio.token" )
+      ( "maximum_supply", "1000000.00 TOK" )
+   );
+}
+
+// test libtester does not enforce limits
+BOOST_AUTO_TEST_CASE( default_limits ) { try {
+   eosvmoc::config eosvmoc_config;
+   limit_not_violated_test(eosvmoc_config);
+} FC_LOG_AND_RETHROW() }
+
+// test VM limits are checked
+BOOST_AUTO_TEST_CASE( vm_limits ) { try {
+   eosvmoc::config eosvmoc_config;
+
+   // disable non-tested limits
+   eosvmoc_config.cpu_limits.rlim_cur       = RLIM_INFINITY;
+   eosvmoc_config.stack_size_limit          = std::numeric_limits<uint64_t>::max();
+   eosvmoc_config.generated_code_size_limit = std::numeric_limits<size_t>::max();
+
+   // set vm_limit to a small value such that it is exceeded
+   eosvmoc_config.vm_limits.rlim_cur = 64u*1024u*1024u;
+   limit_violated_test(eosvmoc_config);
+
+   // set vm_limit to a large value such that it is not exceeded
+   eosvmoc_config.vm_limits.rlim_cur = 128u*1024u*1024u;
+   limit_not_violated_test(eosvmoc_config);
+} FC_LOG_AND_RETHROW() }
+
+// test stack size limit is checked
+BOOST_AUTO_TEST_CASE( stack_limit ) { try {
+   eosvmoc::config eosvmoc_config;
+
+   // disable non-tested limits
+   eosvmoc_config.cpu_limits.rlim_cur       = RLIM_INFINITY;
+   eosvmoc_config.vm_limits.rlim_cur        = RLIM_INFINITY;
+   eosvmoc_config.generated_code_size_limit = std::numeric_limits<size_t>::max();
+
+   // The larget stack size in the test is 104. 
+   // Set stack_size_limit to highest value to verify wasm_execution_error is raised 
```

Comment:
I find this comment confusing. Maybe `// set stack_size_limit less than stack size of test`